client/beefy/src/lib.rs

```diff
@@ -378,20 +384,19 @@ where
 			break state
 		}
 
-		if *header.number() == One::one() {
-			// We've reached chain genesis, initialize voter here.
-			let genesis_num = *header.number();
+		if *header.number() == beefy_genesis {
```

Comment:
What gonna be the output (`state`) of this loop if BEEFY start is scheduled for later? I mean if current best (GRANDPA) finalized header is `100` and BEEFY genesis is set to `200`? Then IIUC the loop will finish with an error when it'll reach the actual (blockchain) genesis and it'll call the `blockchain.expect_header(parent_hash)?;`. And as a result BEEFY gadget won't start.

Reply:
On gadget start, `wait_for_runtime_pallet()` runs/sleeps asynchronously, checking on each GRANDPA finality notification if gadget should start. This initialization code is called only after `GRANDPA final >= BEEFY genesis` as can be seen [here](https://github.com/acatangiu/substrate/blob/c4b6495eb74c15ebc9f97f7c3fcd53f5911706a0/client/beefy/src/lib.rs#L457).

[Only then](https://github.com/acatangiu/substrate/blob/c4b6495eb74c15ebc9f97f7c3fcd53f5911706a0/client/beefy/src/lib.rs#L274-L278), is this initialization code run which will start from said GRANDPA finalized block (known to be >= BEEFY-genesis), and walk back looking for exact BEEFY-genesis block.